docs/user-guide/rug/generators.md

```diff
@@ -1,104 +1,181 @@
+the file `mkdocs.yml` with the generated project's name.
+
+### A Basic Generator
 
 The following generator creates a MkDocs (a documentation builder tool) project
 initialized with default content and title, through an editor.
 
-```typescript linenums="1"
+```typescript
 import { PopulateProject } from "@atomist/rug/operations/ProjectGenerator"
 import { Generator, Tags } from '@atomist/rug/operations/Decorators'
 import { Pattern } from '@atomist/rug/operations/RugOperations'
-import { Project } from '@atomist/rug/model/Core'
+import { Project } from '@atomist/rug/model/Project'
 
 @Generator("NewMkDocsDocumentationProject", "creates a MkDocs project")
 @Tags("markdown", "documentation", "mkdocs")
 class NewMkDocsDocumentationProject implements PopulateProject {
 
-    @Parameter({description: "name of the default document", pattern: Pattern.any})
+    @Parameter({description: "name of the default page", pattern: Pattern.any})
     defaultdoc: string = "index.md"
 
     populate(project: Project) {
         // set the created project's name in the original source
         project.findFile("mkdocs.yml").replace("<NAME>", project.name())
 
+        // create the docs directory
+        project.addDirectory("docs")
+
         // call an editor to add a default document
-        project.editWith("AddMarkdownDocument", {
+        project.editWith("AddMarkdownDocumentToMkDocsProject", {
             "name": "docs/" + this.defaultdoc,
             "title": "My new project",
             "content": "Hi there!"
         })
     }
 }
-export const generator = new NewMkDocsDocumentationProject()
+export let generator = new NewMkDocsDocumentationProject()
 ```
 
-## Run Generators
+Please see below for a [detailed description](#what-is-a-generator-made-of)
+of this generator's structure.
 
-When applied, a generator follows the next steps:
+## Search for Generators
 
-1.  Copy the content of the Rug project, excluding the `.atomist`
-    directory, into a target directory
-2.  Applies the generator's function against the filled target
-    directory to tune its content
+### Using the Atomist bot
 
-Let's assume the following Rug generator project which
+To see the list of available generators to your team, ask the bot for them:
 
 ```
-mkdocs-generator
-    .atomist
-        generators
-            NewMkDocsDocumentationProject.ts
-        tests
-    docs
-    mkdocs.yml
-    requirements.txt
-    README.md
+@atomist generators
 ```
 
-That project itself is a valid [MkDocs][mkdocs] (a documentation builder tool)
-project that can be run in its own right.
+This will return a short list of generators, you can refine your search by
+specifying filter words to the command. For instance, to search for generators
+for Java and Spring:
+
+```
+@atomist java spring
+```
 
-To create a project from this Rug generator, you could run the following
-[rug][cli] command:
+### Using the CLI
```

Comment:
@Lawouach the characters between the hash marks and the capital "U" is ASCII 160, a "non-breaking space". What editor are you using that is inserting these characters? Any way to turn it off?